src/browser/keyboard.js

```diff
@@ -426,11 +442,47 @@ function KeyboardAdapter(bus)
         }
         //console.log("Key: " + code.toString(16) + " from " + chr.toString(16) + " down=" + keydown);
 
-        if(code > 0xFF)
+        if(platfom_windows)
         {
-            // prefix
-            send_to_controller(code >> 8);
-            send_to_controller(code & 0xFF);
+            send_to_controller_filter_altgr(code, e.getModifierState && e.getModifierState("AltGraph"));
+        }
+        else
+        {
+            send_to_controller(code);
+        }
+    }
+
+    function send_to_controller_filter_altgr(code, altgr_key)
+    {
+        // Remove ControlLeft from key sequence [ControlLeft, AltRight] when
+        // AltGr-key is pressed or released.
+        //
+        // NOTE: altgr_key is false for the 1st key (ControlLeft-Down), becomes
+        // true with the 2nd (AltRight-Down) and stays true until key AltGr
+        // is released (AltRight-Up).
+        //
+        // Scancodes (note: bit 0x80 indicates Key-Up):
+        // - 0x1D, 0x9D: ControlLeft-Down, ControlLeft-Up
+        // - 0xE038, 0xE0B8: AltRight-Down, AltRight-Up
+
+        if(deferred_code)
+        {
+            clearTimeout(deferred_timeout_id);
+            if(!(altgr_key && ((deferred_code === 0x1D && code === 0xE038) || (deferred_code === 0x9D && code === 0xE0B8))))
+            {
+                send_to_controller(deferred_code);
+            }
+            deferred_code = 0;
+        }
+
+        if(code === 0x1D || code === 0x9D)
+        {
+            // defer ControlLeft-Down/-Up until the next invocation of this method or 10ms have passed, whichever comes first
+            deferred_code = code;
+            deferred_timeout_id = setTimeout(() => {
+                send_to_controller(deferred_code);
+                deferred_code = 0;
+            }, 10);
```

Comment:
I think this logic should be moved earlier in the chain, into `handler` before `translate`. That way we don't have to pass through the event object, `keys_pressed` doesn't get infected, and we can use `e.code === "AltLeft"` directly.